packages/block-library/src/query/block.json

```diff
@@ -0,0 +1,21 @@
+{
+	"name": "core/query",
+	"category": "layout",
+	"attributes": {
+		"id": {
+			"type": "number"
+		},
+		"query": {
+			"type": "object",
+			"default": {
+				"per_page": 3,
+				"pages": 1,
+				"offset": 0
+			}
+		}
+	},
+	"providesContext": {
+		"id": "id",
+		"query": "query"
```

Comment:
I guess it's a good demonstration of how we probably won't be able to expect other developers to do it either, but part of the reason for the object form was to emphasize the desire for namespacing of context properties, i.e. `core/id`, `core/query`.

https://github.com/WordPress/gutenberg/blob/master/docs/designers-developers/developers/block-api/block-context.md#providing-block-context

Another highlight is that "id" as a name in context is hugely ambiguous, without also being in some way associated to the fact that it's provided by the context block.

Something like `core/queryId` ?

Reply:
Agreed it should be `queryId`, updated.

But, I thought we are dropping the `core` prefix. It's redundant. Plugins have to prefix things anyways, so anything un-prefixed should be assumed to be from `core`.